src/org/zaproxy/zap/extension/scripts/ScriptsTreeCellRenderer.java

```diff
@@ -55,6 +56,13 @@
 	private static final ImageIcon MISSING_ENGINE_ICON = 
 			new ImageIcon(ScriptsTreeCellRenderer.class.getResource(RESOURCE_ROOT + "script-missing-engine.png"));
 
+	private static final ImageIcon JAVASCRIPT_ICON = 
+			new ImageIcon(DisplayUtils.class.getResource(CORE_ROOT + "javascript.png"));
```

Comment:
Could use `ScriptsTreeCellRenderer`.

Reply:
In this case no - its one of the icons included in the core ;)

Reply:
True, but using the other class should work too (it ends up using core class loader to find resources).